### DISCUSSION TITLE
MIOpen Support windows and Mainstream cards

### DISCUSSION THREAD

**Discussion by johnnynunez** [ROCmSoftwarePlatform/MIOpen#2375] [Q&A]
I want to open this discussion to give priority to these two requests: Windows compatibility and RDNA compatibility.

**Comment by atamazov** [comment]
@johnnynunez I highly recommend to not mix two different topics (Windows and RDNA) in the same discussion.

Anyway, the title is very promising, and I believe that you have something substantial to tell or advise the team on these topics, so please do not hesitate to share it with the team.

**Comment by johnnynunez** [comment]
> @johnnynunez I highly recommend to not mix two different topics (Windows and RDNA) in the same discussion.
> 
> Anyway, the title is very promising, and I believe that you have something substantial to tell or advise the team on these topics, so please do not hesitate to share it with the team.

Just imagine the thousands of RDNA GPU's that are in the developer's house. It would encourage much faster development for windows and Linux if there was native RDNA support.

**Comment by scarsty** [comment]
I have built a Windows edition of MIOpen.dll with branch uif2-initial, maybe you can try some benchmarks with it.

MIOPEN_USE_MLIR, MIOPEN_ENABLE_AI_KERNEL_TUNING, MIOPEN_ENABLE_AI_IMMED_MODE_FALLBACK are all Off now. And very many tricks are applied.

<https://drive.google.com/file/d/1OFey0Zr3Hi4xiT6NaRj35h67TBxJ6ZVO/view?usp=sharing>

**Reply by johnnynunez** [reply]
@scarsty with all changes applied, can you compile it?

**Reply by scarsty** [reply]
Miopen itself is not hard to build on Windows, but some dependent libraries are very difficult, so I closed some options. With this dll, 7900xtx is about 2X slower than 2080. Have you tested it on Linux?

**Reply by johnnynunez** [reply]
On linux it works fine for me. I mean that in windows more than 40 PR have been closed since your compilation. I was saying this in case you had compiled on January 18 with all the new code and for rocm6.

**Reply by scarsty** [reply]
Because of the bad speed of my result, I have been waiting for an official version of miopen for months. But yesterday I found the download of rocm-windows have been 404, I think we can say goodbye to it.

**Reply by Sabrewarrior** [reply]
https://www.amd.com/en/developer/resources/rocm-hub/hip-sdk.html
It's been moved but the link is not updated. Could you share the script or environment variables you set to build MIOpen?

**Reply by scarsty** [reply]
It is great, I will try this next week.

**Reply by scarsty** [reply]
Where I can get rocm 6 for Windows? I found the link posted by @Sabrewarrior has also been 404.

**Reply by Sabrewarrior** [reply]
Wow...wasn't expecting that link to be broken before fixing the actual link. There is no ROCm 6 for Windows yet.

Windows HIP SDK 
5.7.1: https://download.amd.com/developer/eula/rocm-hub/AMD-Software-PRO-Edition-23.Q4-Win10-Win11-For-HIP.exe
5.5.1: https://download.amd.com/developer/eula/rocm-hub/AMD-Software-PRO-Edition-23.Q3-Win10-Win11-For-HIP.exe

Lots of open issues regarding this on ROCm: https://github.com/ROCm/ROCm/issues/2794 https://github.com/ROCm/ROCm/issues/2778 https://github.com/ROCm/ROCm/issues/2841 https://github.com/ROCm/ROCm/issues/2841

**Reply by lshqqytiger** [reply]
Could you recompile MIOpen.dll with ROCm 5.7.1 for Windows? I have older one which is built with ROCm 5.5.x, but I can't load it with ROCm 5.7.1 because it depends on hiprtc0505.dll that doesn't exist on 5.7.1.

**Reply by lshqqytiger** [reply]
Found one in #2703. Thanks

**Comment by johnnynunez** [comment]
@apwojcik there are like 22 PR opened for windows. Do you know when they will be all to be compatible MiOpen with windows?

**Reply by FeuFeuAngel** [reply]
Is this dead? There is no real project (just template) and the original PR was closed, so really no activity right now. (Cant wait to try my amd card finally)

**Reply by johnnynunez** [reply]
No, they are working 15 closed and 8 still opened. https://github.com/ROCmSoftwarePlatform/MIOpen/pulls?q=is%3Aopen+is%3Apr+label%3Awindows

**Reply by scarsty** [reply]
have all PRs for windows been merged?

**Reply by johnnynunez** [reply]
Only pending one but still is not created, but yes, 40 PRs was merged https://github.com/ROCm/MIOpen/issues/2689

**Comment by johnnynunez** [comment]
My 7900xt is working well with Pytorch but there are still bugs when the memory is full in tensorflow like the system freezes and the graphics stop working. How is the windows support going? I have seen that there are only 4 Pull Request left. @apwojcik

**Comment by scarsty** [comment]
Is there any new progress? I have seen all PRs about windows have been merged.

**Reply by johnnynunez** [reply]
> Is there any new progress? I have seen all PRs about windows have been merged.

I know that they are putting all efforts to do compatible on windows
